### PR TITLE
Issue button to clean chat history

### PR DESCRIPTION
**Pull request by TomasMatarazzo**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

BACKEND:

- I added this endpoint to delete all the conversations.

image

FRONTEND:

- I added a button next to the chats label, it is hidden if you dont have chats. Let me know if this is ok. 
- When you click the button it opens a pop up that asks if you want to delete all the conversations. For this pop up i used a component that you have in the project called Modal.

image

image

- **Why was this change needed?** (You can also link to an open issue here)

https://github.com/arc53/DocsGPT/issues/927

### CODE REVIEW

**Comment by dartpain** [comment]
Thank you for your PR.
I just have one design change request.
Can we put the button that triggers deletion into the Settings  > General menu.
The modal is great!

**Comment by TomasMatarazzo** [comment]
I changed the button to settings

**Comment by TomasMatarazzo** [comment]
Sure
- I changed the button as you said
image
- About the backend , i thought the same at first. Now all the users are local what i did is hardcode the user_id to local idk if this the correct approach, let me know
image

**Comment by dartpain** [comment]
Hey, there is a quick error on Vercel, please check it out, thank you.

TS2741: Property 'modalState' is missing in type '{ apiKey: string; chunks: any; selectedDocs: any; prompt: any; conversations: null; sourceDocs: { location: string; language: string; name: string; version: string; date: string; description: string; docLink: string; fullName: string; model: string; }[]; }' but required in type '{ apiKey: string; prompt: { name: string; id: string; type: string; }; selectedDocs: { location: string; name: string; language: string; version: string; description: string; fullName: string; date: string; docLink: string; model: string; } | null; chunks: string; sourceDocs: { ...; }[] | null; conversations: { ...;...'.

**Comment by TomasMatarazzo** [comment]
Let me know if now works. Thanks

File: frontend/src/Navigation.tsx [hunk 1]
```diff
@@ -92,16 +98,28 @@ export default function Navigation({ navOpen, setNavOpen }: NavigationProps) {
       fetchConversations();
     }
   }, [conversations, dispatch]);
+
   async function fetchConversations() {
     return await getConversations()
       .then((fetchedConversations) => {
         dispatch(setConversations(fetchedConversations));
+        console.log(conversations);
```
Comment (dartpain) [inline frontend/src/Navigation.tsx:106]: Please remove console log

**Review by dartpain — CHANGES_REQUESTED** [review]
Just a tiny comment to remove console.log statement and we are happy to merge

**Review by dartpain — APPROVED** [review]

**Review by dartpain — APPROVED** [review]

**Comment by dartpain** [comment]
Thank you so much!

**Pull request merged**